recipes/LibriSpeech/ASR/CTC/train_with_wav2vec_k2.py

```diff
@@ -462,6 +462,25 @@ def text_pipeline(wrd):
         lexicon,
         device=asr_brain.device,
     )
+
+    if hparams["decoding_method"] in ["1best", "onebest"]:
+        arpa_path = Path(hparams["lm_dir"]) / hparams["G_arpa_name"]
+        fst_output_path = Path(hparams["lm_dir"]) / hparams["G_fst_output_name"]
+        logger.info(
+            f"Decoding method {hparams['decoding_method']} with following files: "
+            f"arpa: {arpa_path}, fst: {fst_output_path}."
+        )
+    elif hparams["decoding_method"] == "whole-lattice-rescoring":
+        arpa_path = Path(hparams["lm_dir"]) / hparams["G_rescoring_arpa_name"]
+        fst_output_path = (
+            Path(hparams["lm_dir"]) / hparams["G_rescoring_fst_output_name"]
+        )
+        logger.info(
+            f"Decoding method {hparams['decoding_method']} with following files: "
+            f"arpa: {arpa_path}, fst: {fst_output_path},"
+            f"and a rescoring_lm_scale of {hparams['rescoring_lm_scale']}."
+        )
+
```

Comment:
This is already logged in [lattice_decoder.py](https://github.com/speechbrain/speechbrain/pull/2133/files#diff-4442b0184eecf570e7370b9933ad8b67ecf482176d32ac4fd48edc74b8a11e8c) and `utils.load_G`.

```log
 $~ python3 train_with_wav2vec_k2.py hparams/train_hf_wav2vec_k2.yaml --data_folder /data/pchampio/Librispeech/LibriSpeech

Some weights of Wav2Vec2Model were not initialized from the model checkpoint at facebook/wav2vec2-large-960h-lv60-self and are newly initialized: ['wav2vec2.masked_spec_embed']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
speechbrain.lobes.models.huggingface_transformers.huggingface - Wav2Vec2Model is frozen.
speechbrain.core - Beginning experiment!
speechbrain.core - Experiment folder: results/train_wav2vec2_char_k2/1111
librispeech_prepare - Skipping preparation, completed in previous run.
lm/3-gram.arpa.gz exists. Skipping download
lm/3-gram.pruned.1e-7.arpa.gz exists. Skipping download
lm/3-gram.pruned.3e-7.arpa.gz exists. Skipping download
lm/4-gram.arpa.gz exists. Skipping download
speechbrain.k2_integration.prepare_lang - Skipping lang preparation of 'results/train_wav2vec2_char_k2/1111/lang'. Set 'caching: False' in the yaml if this is not what you want.
speechbrain.core - Info: auto_mix_prec arg from hparam file is used
speechbrain.core - Info: ckpt_interval_minutes arg from hparam file is used
speechbrain.core - 2.1M trainable parameters in ASR
speechbrain.k2_integration.lexicon - Loading compiled results/train_wav2vec2_char_k2/1111/lang/L.pt
speechbrain.k2_integration.lexicon - Loading compiled results/train_wav2vec2_char_k2/1111/lang/Linv.pt
HERE --- > speechbrain.k2_integration.utils - Loading 'lm/G_3_gram.fst.txt' from its cached .pt format. Set 'caching: False' in the yaml if this is not what you want.
speechbrain.k2_integration.graph_compiler - Arc sorting L
speechbrain.k2_integration.lexicon - Loading compiled results/train_wav2vec2_char_k2/1111/lang/L_disambig.pt
speechbrain.k2_integration.graph_compiler - Loading HLG 'results/train_wav2vec2_char_k2/1111/.HLG_3216585102610891.pt' from its cached .pt format. Set 'caching: False' in the yaml if this is not what you want.
HERE --- > speechbrain.k2_integration.lattice_decoder - Decoding method: one-best-decoding
speechbrain.utils.checkpoints - Loading a checkpoint from results/train_wav2vec2_char_k2/1111/save/CKPT+2023-09-04+18-28-37+00
```

Reply:
Yes, I saw that, but we got this error during the pre-commit check.

> Checking recipes/LibriSpeech/ASR/CTC/hparams/train_hf_wav2vec_k2.yaml...
	ERROR: variable "G_rescoring_arpa_name" not used in recipes/LibriSpeech/ASR/CTC/train_with_wav2vec_k2.py!
	ERROR: variable "G_arpa_name" not used in recipes/LibriSpeech/ASR/CTC/train_with_wav2vec_k2.py!
	ERROR: variable "rescoring_lm_scale" not used in recipes/LibriSpeech/ASR/CTC/train_with_wav2vec_k2.py!
	ERROR: variable "G_rescoring_fst_output_name" not used in recipes/LibriSpeech/ASR/CTC/train_with_wav2vec_k2.py!
	ERROR: variable "G_fst_output_name" not used in recipes/LibriSpeech/ASR/CTC/train_with_wav2vec_k2.py!

Do you know of any ideas on how to solve it?

Reply:
make sure that unused variables are not part of the yaml, otherwise you will get a pre-commit error.

Reply:
@Adel-Moumen those variables are used in lattice_loader.py but not train.py.
The pre-commit only checks for train.py usage but not lattice_loader.py

Reply:
It should detect when you are passing a key. I will have a double check latter today :)